lib/main.dart

```diff
@@ -25,21 +37,60 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return GetMaterialApp(
-      locale: const Locale('pl', 'PL'),
-      translations: LocaleString(),
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
-        scaffoldBackgroundColor: Colors.white,
+    return MultiRepositoryProvider(
+      providers: [
+        RepositoryProvider<AuthRepository>(
+          create: (context) => AuthRepository(
+            firebaseFirestore: FirebaseFirestore.instance,
+            firebaseAuth: FirebaseAuth.instance,
+          ),
+        ),
+        RepositoryProvider<ProfileRepository>(
+          create: (context) => ProfileRepository(
+            firebaseFirestore: FirebaseFirestore.instance,
+          ),
+        ),
+      ],
+      child: MultiBlocProvider(
+        providers: [
+          BlocProvider<AuthBloc>(
+            create: (context) => AuthBloc(
+              authRepository: context.read<AuthRepository>(),
+            ),
+          ),
+          BlocProvider<SignInCubit>(
+            create: (context) => SignInCubit(
+              authRepository: context.read<AuthRepository>(),
+            ),
+          ),
+          BlocProvider<SignUpCubit>(
+            create: (context) => SignUpCubit(
+              authRepository: context.read<AuthRepository>(),
+            ),
+          ),
+          BlocProvider<UserCubit>(
+            create: (context) => UserCubit(
+              profileRepository: context.read<ProfileRepository>(),
+            ),
+          ),
+        ],
```

Comment:
What do you think about extract it to seperate widgets called bloc wrapper, repository wrapper or something similiar? Your main will be much cleaner, that's the good practise and using e.g. in foodstuff :)

Reply:
Ok Yep there is more cleaner after that